src/prefect/projects/steps/build.py

```diff
@@ -0,0 +1,72 @@
+"""
+Core set of steps for specifying a Prefect project build step.
+"""
+import os
+import sys
+from pathlib import Path
+
+import pendulum
+
+from prefect.docker import build_image, docker_client, get_prefect_image_name
+from prefect.utilities.slugify import slugify
+
+
+def build_docker_image(
+    image_name: str,
+    dockerfile: str = None,
+    auto_build: bool = False,
+    tag: str = None,
+    push: bool = True,
+) -> dict:
+    if auto_build:
+        lines = []
+        base_image = get_prefect_image_name(prefect_version="2.8.5")
```

Comment:
We should be able to let this function dynamically determine the corresponding image name.
```suggestion
        base_image = get_prefect_image_name()
```